### PR TITLE
Collect all transformation errors reading a PB model into detached model

### PR DESCRIPTION
**Pull request by christoph-wulf-moia**
* Previously the implementation failed fast at the first error

#### Has the version number been increased?
 - [x] Yes! (or not but it was intended to not increase it)

### CODE REVIEW

File: src/main/scala/io/moia/protos/teleproto/PbResult.scala [hunk 1]
```diff
@@ -99,4 +105,13 @@ object PbFailure {
 
   def apply(message: String): PbFailure =
     apply("", message)
+
+  /**
+    * Collects and combines all the errors of all failures.
+    */
+  def combine(results: PbResult[_]*): PbFailure =
```
Comment (aquamatthias) [inline src/main/scala/io/moia/protos/teleproto/PbResult.scala]: Not 100% sure: if all results would be PbSuccess the result should be PbSuccess or?

File: src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala [hunk 2]
```diff
@@ -134,56 +136,97 @@ object ReaderImpl {
...
+              q"""${termSymbol.name}.flatMap { case ${termSymbol.name} => $restTransformed }"""
+          }
+      }
 
-          val path = q""""/"+${termSymbol.name.decodedName.toString}"""
+    /**
+      * Constructs an expression `PbFailure.combine(monadicParameters..)` for all transformed parameters (could fail).
+      */
+    def combineErrors(parameters: List[(TermSymbol, MatchingParam[Type, Tree])]): Tree = {
+      val monadicValues =
```
Comment (aquamatthias) [inline src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: Is this flatMap really aggregating monadic values?

File: src/main/scala/io/moia/protos/teleproto/PbResult.scala [hunk 3]
```diff
@@ -99,4 +105,13 @@ object PbFailure {
...
+    PbFailure(results.flatMap {
+      case PbSuccess(_)      => Seq.empty
```
Comment (GodPlaysChess) [inline src/main/scala/io/moia/protos/teleproto/PbResult.scala]: I don't understand how this works.
`results` is of type `PbResult`.
`flatMap` is of type `A => PbResult`.

you pass something of type `PbResult => Seq` to the `flatMap`.
Why does it even compile?

File: src/main/scala/io/moia/protos/teleproto/PbResult.scala (hunk 1 above)
Comment (christoph-wulf-moia) [reply src/main/scala/io/moia/protos/teleproto/PbResult.scala]: I added a comment that explains the usage. It cannot be a `PbSuccess` without a combining method and that's not the intention of that helper.

File: src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala (hunk 2 above)
Comment (christoph-wulf-moia) [reply src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: I call it `convertedValues` now.

Comment (christoph-wulf-moia) [reply src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: Needed a specific name for `PbResult[T]` instead of just a `T` that can be copied.

**Pull request merged**